samples/showcase/src/main/java/com/facebook/fresco/samples/showcase/imageformat/svg/SvgDecoderExample.java

```diff
@@ -50,13 +55,23 @@ public int getHeaderSize() {
 
     @Nullable
     @Override
-    public ImageFormat determineFormat(byte[] headerBytes, int headerSize) {
+    public ImageFormat determineFormat(byte[] headerBytes, int headerSize, InputStream is)
+        throws IOException {
       if (headerSize < getHeaderSize()) {
         return null;
       }
       if (ImageFormatCheckerUtils.startsWithPattern(headerBytes, HEADER)) {
         return SVG_FORMAT;
       }
+
+      // TODO: (1) Determine if passing the InputStream to this method is the best way to achieve this result.
+      // TODO: (2) Determine if searching the whole array of bytes in the stream is best practice, according to the repository authors.
+      int indexOfPattern = ImageFormatCheckerUtils.containsPattern(ByteStreams.toByteArray(is), HEADER);
+      if (indexOfPattern != -1) {
+        PooledByteBufferInputStream.byteOffset = indexOfPattern;
```

Comment:
If you'd really want the image format checker to have side effects: You probably want to consume the first `indexOfPattern` bytes instead.